### PR TITLE
Mccalluc/update url

### PR DESCRIPTION
**Pull request by mccalluc**
sorts and filters are preserved in the url, and the url is consulted during page load.

Somewhat concerned about code organization.

### CODE REVIEW

**Comment by mccalluc** [comment]
Fix #2005 and fix #1959

**Comment by mccalluc** [comment]
Failed build ends with 
```
.2017-08-31 13:21:28 DEBUG    pyvirtualdisplay.abstractdisplay:90 redirect_display() - DISPLAY=:1001
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

**Comment by hackdna** [comment]
Same as #2067

**Comment by scottx611x** [comment]
@hackdna I don't think this error is the same as #2067. It looks more like a spurious Travis build error to me.

File: refinery/ui/source/js/user-file-browser/ctrls/user-file-browser-filters-ctrl.js [hunk 1]
```diff
@@ -47,20 +71,36 @@
...
-        promise.resolve();
-      }, function () {
-        $log.error('/files/ request failed');
-        promise.reject();
-      });
-    };
+    var sort;
+    var direction;
+    angular.forEach($location.search(), function (values, key) {
+      if (key === 'sort') {
```
Comment (scottx611x) [inline refinery/ui/source/js/user-file-browser/ctrls/user-file-browser-filters-ctrl.js]: I saw variables set for `'sort'` and `'direction'` above? Could they be reused?

**Comment by hackdna** [comment]
The test run fails to proceed after the same output line in both cases:
```
.2017-08-31 13:21:28 DEBUG    pyvirtualdisplay.abstractdisplay:90 redirect_display() - DISPLAY=:1001
```

**Comment by scottx611x** [comment]
@hackdna In #2067 The test proceeds to fail, but here the Travis build is hanging and not doing anything for 10 minutes. I think these two aren't necessarily unrelated, but aren't the same. Especially if #2067 is consistently reproducible.

**Comment by hackdna** [comment]
Right, the test run in Vagrant is also hanging and doing nothing.

**Pull request merged**